File: wrapped/funcs/f_66.py
def is_spam(message):
    import re

    # Check for typical spam phrases and symbols
    spam_phrases = ["상한가", "특별 할인", "무료수신거부", "%", "MOU", "특가", "소문난 주식"]

    for phrase in spam_phrases:
        if phrase in message:
            return True

    # Check for excessive use of special characters
    special_chars = re.findall(r"[!@#$%^&*()\-_=+[\]{};:\'\"|,<.>/?]+", message)
    if len(special_chars) > 5:
        return True

    # Check for suspicious URLs
    urls = re.findall(r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+", message)
    if len(urls) > 1:
        return True

    # Check for excessive use of numbers
    numbers = re.findall(r"[0-9]+", message)
    if len(numbers) > 3:
        return True

    # Check for non-Korean characters
    non_korean = re.findall(r"[^ㄱ-하-ㅣ가-힣\s]+", message)
    if len(non_korean) > 5:
        return True

    return False

File: wrapped_kotlin_ported/funcs/f_66.kt
fun isSpam(message: String): Boolean {
    val spamPhrases = listOf("상한가", "특별 할인", "무료수신거부", "%", "MOU", "특가", "소문난 주식")

    spamPhrases.forEach {
        if (it in message) return true
    }

    val specialChars = Regex("[!@#$%^&*()\\-_=+\\[\\]{};:\'\"|,<.>/?]+").findAll(message).toList()
    if (specialChars.size > 5) return true

    val urls = Regex("http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\\\(\\\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+").findAll(message).toList()
    if (urls.size > 1) return true

    val numbers = Regex("[0-9]+").findAll(message).toList()
    if (numbers.size > 3) return true

    val nonKorean = Regex("[^ㄱ-하-ㅣ가-힣\\s]+").findAll(message).toList()
    if (nonKorean.size > 5) return true

    return false
}
